ably/rest_client.go

```diff
@@ -118,13 +117,15 @@ type REST struct {
 	Channels            *RESTChannels
 	opts                *clientOptions
 	successFallbackHost *fallbackCache
+	log                 logger
 }
 
 // NewREST constructs a new REST.
 func NewREST(options ...ClientOption) (*REST, error) {
 	c := &REST{
 		opts: applyOptionsWithDefaults(options...),
 	}
+	c.log = logger{l: c.opts.LogHandler}
```

Comment:
Just curious, are we passing any logger context here? as in this logger belongs to `RestClient`, so everytime it logs stuff, 
it will be prefixed with `RestClient`

Reply:
And it should be applicable at every other place, where we are initializing domain specific logger

Reply:
That's what #323 is about, this PR just cleans up what's already there.

Reply:
#323 is closed now but the task will be done in a further PR.